geopandas/geodataframe.py

```diff
@@ -329,6 +329,28 @@ def crs(self, value):
                 # column called 'geometry' without geometry
                 self._crs = None if not value else CRS.from_user_input(value)
 
+    def __setstate__(self, state):
+        # overriding DataFrame method for compat with older pickles (CRS handling)
+        if isinstance(state, dict):
+            if "_metadata" in state and "crs" in state["_metadata"]:
+                metadata = state["_metadata"]
+                metadata[metadata.index("crs")] = "_crs"
+            if "crs" in state and "_crs" not in state:
+                crs = state.pop("crs")
+                state["_crs"] = CRS.from_user_input(crs) if crs is not None else crs
+
+        super().__setstate__(state)
+
+        # for some versions that didn't yet have CRS at array level -> crs is set
+        # at GeoDataFrame level with '_crs' (and not 'crs'), so without propagating
+        # to the GeoSeries/GeometryArray
+        try:
+            if self.crs is not None:
+                if self.geometry.crs is None:
```

Comment:
I guess we should check for GeometryArray crs here, not GeoSeries, because that inherited GeoDataFrame's CRS in pre 0.8.

Reply:
Normally, GeoSeries should get the crs from the array, and not the dataframe. So I *think* it shouldn't matter, but will check (also doesn't harm to be more explicit)

(but it's from the geoseries missing the crs that I noticed it, I think)